src/actions/walletActions.js

```diff
@@ -21,46 +21,48 @@ async function _createCloudWallet (password: string, progress: ?Function) {
   // the following creation process should never fail (which blocks the onboarding process),
   // warn silently for any errors
   try {
-    let newAccountList = []
-    for (const { cryptoType, disabled } of walletCryptoSupports['drive']) {
-      if (!disabled) {
-        let account
-        let accountData
-        let _wallet = createWallet({ walletType: 'drive' })
-
-        if (ethereumBasedAccountData && ['ethereum', ...erc20TokensList].includes(cryptoType)) {
-          // share the same privateKey for ethereum based coins
-          await _wallet.newAccount('Wallet', cryptoType, {
-            privateKey: ethereumBasedAccountData.privateKey
-          })
-          account = _wallet.getAccount()
-          accountData = account.getAccountData()
-        } else {
-          await _wallet.newAccount('Wallet', cryptoType, {
-            getPrefilled: true
-          })
-          account = _wallet.getAccount()
-          accountData = account.getAccountData()
-          if (['ethereum', ...erc20TokensList].includes(cryptoType)) {
-            ethereumBasedAccountData = JSON.parse(JSON.stringify(accountData))
-          }
-        }
+    let newAccountList = await Promise.all(
+      walletCryptoSupports['drive']
+        .filter(e => !e.disabled)
+        .map(async ({ cryptoType }) => {
+          let account
+          let accountData
+          let _wallet = createWallet({ walletType: 'drive' })
 
-        if (accountData.cryptoType === 'bitcoin') {
-          walletFileData[accountData.hdWalletVariables.xpub] = JSON.parse(
-            JSON.stringify(account.getAccountData())
-          )
-          await account.encryptAccount(password)
-          encryptedWalletFileData[accountData.hdWalletVariables.xpub] = account.getAccountData()
-        } else {
-          walletFileData[accountData.address] = JSON.parse(JSON.stringify(account.getAccountData()))
-          await account.encryptAccount(password)
-          encryptedWalletFileData[accountData.address] = account.getAccountData()
-        }
+          if (ethereumBasedAccountData && ['ethereum', ...erc20TokensList].includes(cryptoType)) {
+            // share the same privateKey for ethereum based coins
+            await _wallet.newAccount('Wallet', cryptoType, {
+              privateKey: ethereumBasedAccountData.privateKey
+            })
+            account = _wallet.getAccount()
+            accountData = account.getAccountData()
+          } else {
+            await _wallet.newAccount('Wallet', cryptoType, {
```

Comment:
If the first cryptoType encountered is DAI, and ethereumBasedAccountData is not defined yet. Since you make the change here https://github.com/ventureum/Chainsfr/pull/1630/files#diff-3ea1faa9fab9aa60c2e95abb20f9fe7fR73, it probably wont fetch prefilled account.